### DISCUSSION TITLE
.NET Support

### DISCUSSION THREAD

**Discussion by iainmckay** [google/filament#5923] [General]
Hey guys, I wanted to use Filament in my project and saw that there were no .NET bindings so I went off and implemented them. 

I opted for a straight manual P/Invoke approach instead of something automated (e.g. SWIG) as I have no experience with those tools and I wanted to get something running quickly and that suited my needs.

There are two projects:

https://github.com/chicken-with-lips/filament/tree/dotnet/dotnet/filament
This is my fork of `filament` and contain what is essentially a copy of the JNI bindings tweaked to run in .NET.

https://github.com/chicken-with-lips/filament.net
This contains two .NET projects. `Filament.Native` which are straight P/Invoke bindings and `Filament` which is a more "user friendly" layer on top of `Filament.Native`. These both target .NET Core. This repository also contains several ported samples.

There is still much to do as I only exposed enough to get the samples running (which is a fair bit of coverage) and I've only built this for Linux. However, I've successfully integrated Filament in to my project using these bindings.

As my time is a little short at the moment I wanted to release this as at the very least should give other people a starting point that are interested. My time should free up again soon and I'll be adding the missing README, inline documentation and updating to the latest version of Filament (currently 1.9.12 is supported).

Feedback is more than welcome as are PR's.

**Comment by vpenades** [comment]
Any update on this?
